### ISSUE TITLE
Deleting one document also causes other documents to disappear from searches

### ISSUE THREAD

**Issue by lucaong** [nearform/lyra#40]
**Describe the bug**
The bug occurs in `lyra-0.0.1-beta-13`.

Sorry for opening several bugs, I was trying to compare Lyra with [MiniSearch](https://github.com/lucaong/minisearch) and ran into these issues.

**To Reproduce**

```javascript
const db = new Lyra({
  schema: {
    txt: "string",
  }
})

await db.insert({ txt: 'abc' })
//=> { id: 'J-LiSKu45O4d1phnVWvkc' }
await db.insert({ txt: 'abc' })
//=> { id: 'yx-s9o0-8I4uWcPsMyOWL' }
await db.insert({ txt: 'abcd' })
{ id: 'XoKJ6uWhRkftAt06UsdKo' }

await db.search({ term: 'abc', exact: true })
/* Returns what expected:
{
  elapsed: '814μs',
  hits: [
    { id: 'J-LiSKu45O4d1phnVWvkc', txt: 'abc' },
    { id: 'yx-s9o0-8I4uWcPsMyOWL', txt: 'abc' },
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined
  ],
  count: 2
}
*/

// Delete one of the two documents containing term `"abc"`
db.delete('yx-s9o0-8I4uWcPsMyOWL')

await db.search({ term: 'abc', exact: true })
/* Returns no results, even if one doc with "abc" should still be in the index:
{
  elapsed: '770μs',
  hits: [
    undefined, undefined,
    undefined, undefined,
    undefined, undefined,
    undefined, undefined,
    undefined, undefined
  ],
  count: 0
}
*/
```

**Expected behavior**

Deleting one document should have no effect on other documents.

**Comment by lucaong** [comment]
I don't think this is fixed by #41 . The test in that PR has an issue that leads to a false positive.

**Comment by lucaong** [comment]
Check my comment on #45, as it fixes the issue but leaves dead code and possible performance concerns.
